src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -383,100 +397,281 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
+            T newParamNode,
+            int originalIndex,
+            int newIndex,
+            int reorderedParametersCount) where T : SyntaxNode
         {
-            var oldTrivia = oldArgument.GetLeadingTrivia();
-            var oldOnlyHasWhitespaceTrivia = oldTrivia.All(t => t.IsKind(SyntaxKind.WhitespaceTrivia));
+            // Copy whitespace trivia from original position.
+            newParamNode = TransferLeadingWhitespaceTrivia(newParamNode, list[newIndex]);
+
+            // The separator associated with the node should have its trivia transferred over as well. If we're looking at the last node, then there is no associated separator.
+            var originalSeparator = originalIndex < list.Count - 1 ? list.GetSeparator(originalIndex) : default;
+            var newSeparator = newIndex < reorderedParametersCount - 1 ? list.GetSeparator(newIndex) : default;
+
+            // If the associated node is not switching positions, we don't need to do any work.
+            if (originalSeparator == newSeparator)
+            {
+                return (newParamNode, newSeparator);
+            }
+
+            // If we have a comment of format '//', we switch the trivia notation to /* */.
+            newParamNode = ChangeNodeSingleLineCommentsToMultiLine(newParamNode);
+            originalSeparator = ChangeSeparatorSingleLineCommentsToMultiLine(originalSeparator);
+
+            // Case 1: The current node in the updated configuration is the last parameter.
+            // Append any trailing trivia from the original separator to the trailing trivia of the new node, as long as it's not all whitespace. We also want to ignore end-of-line trivia.
+            if (newSeparator.IsKind(SyntaxKind.None))
+            {
+                if (originalSeparator.TrailingTrivia.Any(trivia => !trivia.IsKind(SyntaxKind.WhitespaceTrivia)))
+                {
+                    newParamNode = newParamNode.WithAppendedTrailingTrivia(originalSeparator.TrailingTrivia.Where(trivia => !trivia.IsKind(SyntaxKind.EndOfLineTrivia)));
+                }
+            }
+            // Case 2: The current node in the updated configuration is not the last parameter.
+            else
+            {
+                // We want to transfer over the comments from the original separator to the new separator. However, we also want to preserve the new
+                // signature's formatting. To accomplish this, if the original separator has any comments, we will append them at the end of the new separator's
+                // trailing trivia, but before any end-of-line trivia the new separator may have. Existing comments in the new separator will be removed, as they
+                // will be placed in the correct locations in previous or future iterations of the loop.
+                var originalSeparatorComments = originalSeparator.TrailingTrivia.Where(trivia => trivia.IsKind(SyntaxKind.MultiLineCommentTrivia) || trivia.IsKind(SyntaxKind.WhitespaceTrivia));
```

Comment:
this variable name is weird.  you say it's original comments, but it includes whitespace.